### PR TITLE
:sparkles: Make directory-not-empty error more self-explanatory

### PR DESCRIPTION
**Pull request by justinsb**
When I encountered the error, I had to go to the source to understand
the problem.  Made the message more user-focused.
    
Also skipped printing the allowlist, both to fit into lint line-length
limits and to avoid overwhelming the user.

### CODE REVIEW

File: pkg/plugins/golang/v3/init.go
Comment (Adirio) [inline pkg/plugins/golang/v3/init.go:190]: NIT: I think the file name should be enough
```suggestion
				return fmt.Errorf("target directory is not empty: found existing file %q", info.Name())
```

Comment (justinsb) [reply pkg/plugins/golang/v3/init.go:190]: I started with that actually and switched to path :-) The root cause here was that we were finding files like `.git/HEAD`.  info.Name() prints "HEAD", but path prints ".git/HEAD".

**Comment by justinsb** [comment]
/retest  pull-kubebuilder-e2e-k8s-1-16-2

I can't see how that failure could be related, so rolling the dice again!

**Review by estroz — APPROVED** [review]
/approve

**Comment by Adirio** [comment]
My only question is the one raised in https://github.com/kubernetes-sigs/kubebuilder/pull/1944#issuecomment-759508897
I will `lgtm` and `hold` it, If you consider that we should do it in two PRs just remove the `hold` and it will merge. If you opt into a single PR then close it and update the other PR accordingly.

/lgtm
/hold

File: pkg/plugins/golang/v3/init.go
Comment (camilamacedo86) [reply pkg/plugins/golang/v3/init.go:190]: Should we not still clarifying what files are allowed? See that any file with the prefix (.) is allowed. E.g. .gitignore

Comment (justinsb) [reply pkg/plugins/golang/v3/init.go]: My thought was that maybe the user doesn't care, but it is a good idea to try to reference the idea that some files are allowed, so I added it back!  Thanks!

**Review by estroz — APPROVED** [review]
/lgtm

**Pull request merged**
